### DISCUSSION TITLE
2.16

### DISCUSSION THREAD

**Discussion by gasman** [wagtail/wagtail#7928] [Announcements]
* Added support for Django 4.0
 * Removed support for Django 3.0 and 3.1
 * Removed support for Python 3.6
 * Added persistent IDs for ListBlock items, allowing commenting and improvements to revision comparisons (Matt Westcott, Tidjani Dia, with sponsorship from [NHS](https://www.nhs.uk/))
 * Added Aging Pages report (Tidjani Dia)
 * Added automatic redirect creation feature (Andy Babic, with sponsorship from [The National Archives](https://www.nationalarchives.gov.uk))
 * Added `page_slug_changed` signal for pages (Andy Babic)
 * Add more SketchFab oEmbed patterns for models (Tom Usher)
 * Add collapse option to `StreamField`, `StreamBlock`, and `ListBlock` which will load all sub-blocks initially collapsed (Matt Westcott)
 * Private pages can now be fetched over the API (Nabil Khalil)
 * Added `alias_of` field to the pages API (Dmitrii Faiazov)
 * Add support for Azure CDN and Front Door front-end cache invalidation (Tomasz Knapik)
 * Improved styling of workflow timeline modal view (Tidjani Dia)
 * Add secondary actions menu in edit page headers (Tidjani Dia)
 * Removed WOFF fonts
 * Add system check for missing core Page fields in `search_fields` (LB (Ben Johnston))
 * Improve CircleCI frontend & backend build caches, add automated browser accessibility test suite in CircleCI (Thibaud Colas)
 * Add a 'remember me' checkbox to the admin sign in form, if unticked (default) the auth session will expire if the browser is closed (Michael Karamuth, Jake Howard)
 * When returning to image or document listing views after editing, filters (collection or tag) are now remembered (Tidjani Dia)
 * Improve the visibility of field error messages, in Windows high-contrast mode and out (Jason Attwood)
 * Improve implementations of visually-hidden text in explorer and main menu toggle (Martin Coote)
 * Add locale labels to page listings (Dan Braghis)
 * Add locale labels to page reports (Dan Braghis)
 * Change release check domain to releases.wagtail.org (Jake Howard)
 * Add the user who submitted a page for moderation to the "Awaiting your review" homepage summary panel (Tidiane Dia)
 * When moving pages, default to the current parent section (Tidjani Dia)
 * `admin/expanding_formset.js` has been renamed to `admin/expanding-formset.js` (LB (Ben Johnston))
 * Change docs URL to docs.wagtail.org (Jake Howard)
 * Update links to wagtail.io to point to new domain wagtail.org (Jake Howard)
 * Add borders to TypedTableBlock to help visualize rows and columns (Scott Cranfill)
 * Set default submit button label on generic create views to 'Create' instead of 'Save' (Matt Westcott)
 * Improve display of image listing for long image titles (Krzysztof Jeziorny)
 * Use SVG icons in admin home page site summary items (Jérôme Lebleu)
 * Ensure site summary items wrap on smaller devices on the admin home page (Jérôme Lebleu)
 * Rework Workflow task chooser modal to align with other chooser modals, using consistent pagination and leveraging class based views (Matt Westcott)
 * Implemented a locale switcher on the forms listing page in the admin (Dan Braghis)
 * Implemented a locale switcher on the page chooser modal (Dan Braghis)
 * Implemented the `wagtail_site` template tag for Jinja2 (Vladimir Tananko)
 * Change webmaster to website administrator in the admin (Naomi Morduch Toubman)
 * Added documentation for creating custom submenus in the admin menu (Sævar Öfjörð Magnússon)
 * Choice blocks in StreamField now show label rather than value when collapsed (Jérôme Lebleu)
 * Added documentation to clarify configuration of user-uploaded files (Cynthia Kiser)
 * Change security contact address to security@wagtail.org (Jake Howard)
 * Fix: Accessibility fixes for Windows high contrast mode; Dashboard icons colour and contrast, help/error/warning blocks for fields and general content, side comment buttons within the page editor, dropdown buttons (Sakshi Uppoor, Shariq Jamil, LB (Ben Johnston), Jason Attwood)
 * Fix: Rename additional 'spin' CSS animations to avoid clashes with other libraries (Kevin Gutiérrez)
 * Fix: `default_app_config` deprecations for Django >= 3.2 (Tibor Leupold)
 * Fix: Refresh page from database on create before passing to hooks. Page aliases get correct `first_published_date` and `last_published_date` (Dan Braghis)
 * Fix: Additional login form fields from `WAGTAILADMIN_USER_LOGIN_FORM` are now rendered correctly (Michael Karamuth)
 * Fix: Icon only button styling issue on small devices where height would not be set correctly (Vu Pham)
 * Fix: Add padding to the Draftail editor to ensure `ol` items are not cut off (Khanh Hoang)
 * Fix: Prevent opening choosers multiple times for Image, Page, Document, Snippet (LB (Ben Johnston))
 * Fix: Ensure subsequent changes to styles files are picked up by Gulp watch (Jason Attwood)
 * Fix: Ensure that programmatic page moves are correctly logged as 'move' and not 'reorder' in some cases (Andy Babic)

<hr /><em>This discussion was created from the release <a href='https://github.com/wagtail/wagtail/releases/tag/v2.16'>2.16</a>.</em>

**Comment by itekhi** [comment]
Damn Wagtail, I love you guys! 🔥🔥

**Comment by engineervix** [comment]
Wow! This is phenomenal! A super awesome CMS just keeps getting more and more awesome with each new release! 👏👏👏
